keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/GaussianKDE.java

```diff
@@ -0,0 +1,38 @@
+package io.improbable.keanu.algorithms.variational;
+
+import io.improbable.keanu.algorithms.VertexSamples;
+import io.improbable.keanu.algorithms.mcmc.MetropolisHastings;
+import io.improbable.keanu.distributions.continuous.Gaussian;
+import io.improbable.keanu.distributions.continuous.Uniform;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertexSamples;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.probabilistic.KDEVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.ProbabilisticDouble;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class GaussianKDE {
+
+    public KDEVertex approximate(VertexSamples<DoubleTensor> vertexSamples){
+
+        List<Double> samples = vertexSamples.asList().stream()
+            .map(tensor -> tensor.scalar())
```

Comment:
Does this mean it only works on scalars?

Reply:
Yes, it only works with samples drawn from a univariate distribution.

Reply:
In that case, you should probably throw an exception if the tensor is not scalar. The behaviour for a non-scalar is that it will just return the first value and throw away the rest.

Reply:
Ok, i'll throw an `IllegalArgumentException` here. I also added a test for it.